src/components/About/WarningBanner.js

```diff
@@ -0,0 +1,79 @@
+import _ from 'lodash';
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import {
+  Headline,
+  List,
+  MessageBanner,
+} from '@folio/stripes-components';
+
+import { isVersionCompatible } from '../../discoverServices';
+import css from './About.css';
+
+const WarningBanner = ({
+  interfaces,
+  modules,
+}) => {
+  const [missingModules, setMissingModules] = useState([]);
+  const [incompatibleModule, setIncompatibleModule] = useState([]);
+
+  useEffect(() => {
+    const modulesArray = _.flatten(_.values(modules));
+
+    modulesArray.forEach(module => {
+      const okapiInterfaces = module.okapiInterfaces;
+      if (okapiInterfaces) {
+        Object.keys(okapiInterfaces).forEach(key => {
```

Comment:
I suggest to use reduce to build array of missing and another reduce to build incompatible arrays - and move common logic to separate function